sys/linux/security_yama.txt

```diff
@@ -0,0 +1,22 @@
+# Copyright 2022 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/linux/fcntl.h>
+
+resource fd_yama_ptrace_scope[fd]
+
+# 0 - YAMA_SCOPE_DISABLED
+# 1 - YAMA_SCOPE_RELATIONAL
+# 2 - YAMA_SCOPE_CAPABILITY
+# 3 - YAMA_SCOPE_NO_ATTACH
+modes = "0", "1", "2", "3"
+type yama_mode stringnoz[modes]
+
+yama_ptrace_scope_mode {
+	mode	yama_mode
+	z	const[0, int8]
```

Comment:
This can be just `string[modes]`.
Then you can use it directly in the write call: `data ptr[in, string[modes]]`.
Read does not need to use the exact type, `out` types are not filled with data anyway. Usage of `yama_ptrace_scope_mode` in read just restricts buffer size to 2. For reads it's better to use `data ptr[out, array[int8]]`.